### PR TITLE
Update c2chapel to use c_size_t instead of size_t

### PR DESCRIPTION
**Pull request by bmcdonald3**
In https://github.com/chapel-lang/chapel/pull/19284, size_t was updated to c_size_t in the CTypes module, so this modifies c2chapel to reflect that.

This was discovered when trying to convert the idn2 library with c2chapel, as part of https://github.com/chapel-lang/chapel/issues/20754

### CODE REVIEW

**Comment by lydia-duncan** [comment]
Should we try to make some tests of c2chapel to ensure we notice when future changes break it?

**Comment by bmcdonald3** [comment]
We have investigated that in the past, but, unfortunately, it is pretty difficult to fit c2chapel into the `start_test` framework like that.

In https://github.com/Cray/chapel-private/issues/2512, I was trying to do something like that and wasn't able to fit something in to `start_test` in a way that made sense, so I think that I would put that off for now.

(plus, with Chapel 2.0 on the horizon, hopefully there won't be too many more of these 😄)

File: tools/c2chapel/c2chapel.py [hunk 1]
```diff
@@ -84,7 +84,7 @@
 c2chapel["uintptr_t"]          = "c_uintptr"
 c2chapel["ptrdiff_t"]          = "c_ptrdiff"
 c2chapel["ssize_t"]            = "ssize_t"
```
Comment (bradcray) [inline tools/c2chapel/c2chapel.py:86]: Sorry I didn't catch this before seeing the merge message:  `ssize_t` should similarly be updated to `c_ssize_t`.

Comment (bmcdonald3) [reply tools/c2chapel/c2chapel.py:86]: Wow, I was looking right at that PR that made these changes and still misses this... should be fixed by https://github.com/chapel-lang/chapel/pull/20836

**Pull request merged**